cloudinit/config/schema.py

```diff
@@ -96,21 +101,35 @@ def validate_cloudconfig_schema(config, schema, strict=False):
     # Allow for bytes to be presented as an acceptable valid value for string
     # type jsonschema attributes in cloud-init's schema.
     # This allows #cloud-config to provide valid yaml "content: !!binary | ..."
+
     if hasattr(Draft4Validator, 'TYPE_CHECKER'):  # jsonschema 3.0+
         type_checker = Draft4Validator.TYPE_CHECKER.redefine(
             'string', is_schema_byte_string)
         cloudinitValidator = extend(Draft4Validator, type_checker=type_checker)
     else:  # jsonschema 2.6 workaround
         types = Draft4Validator.DEFAULT_TYPES
-        # Allow bytes as well as string (and disable a spurious
-        # unsupported-assignment-operation pylint warning which appears because
-        # this code path isn't written against the latest jsonschema).
+        # Allow bytes as well as string (and disable a spurious unsupported
+        # assignment-operation pylint warning which appears because this
+        # code path isn't written against the latest jsonschema).
         types['string'] = (str, bytes)  # pylint: disable=E1137
         cloudinitValidator = create(
             meta_schema=Draft4Validator.META_SCHEMA,
             validators=Draft4Validator.VALIDATORS,
             version="draft4",
             default_types=types)
+
+    if strict_metaschema:
+        mymeta = cloudinitValidator.META_SCHEMA
+
+        # this disables bottom-level keys
+        mymeta['additionalProperties'] = False
+
```

Comment:
Is this code necessary or referenced anywhere anymore?

Reply:
Yes, I'll update that to be more obvious. What is getting updated here is actually cloudinitValidator.META_SCHEMA['additionalProperties'].